### PR TITLE
analytics events collected through Analytics singleton instead of Configuration

### PR DESCRIPTION
**Pull request by skomputer**
Also added top_genre() to Work as needed by NYPL-simplified/circulation#276.

### CODE REVIEW

File: analytics.py [hunk 1]
```diff
@@ -22,4 +22,14 @@ def collect_event(self, _db, license_pool, event_type, time=None, **kwargs):
         if not time:
             time = datetime.datetime.utcnow()
         for provider in self.providers:
-            provider.collect_event(_db, license_pool, event_type, time, **kwargs)
\ No newline at end of file
+            provider.collect_event(_db, license_pool, event_type, time, **kwargs)
+
+def format_age_range(r):
```
Comment (leonardr) [inline analytics.py]: This is very similar to Work.target_age_string. Is there a way to reuse/improve that code?

**Comment by leonardr** [comment]
:+1: unless Work.target_age_string can replace format_age_range.

**Comment by leonardr** [comment]
:+1:

File: analytics.py [hunk 2]
```diff
@@ -1,25 +1,48 @@
...
             provider_class = getattr(provider_module, "Provider")
             analytics_providers.append(provider_class.from_config(config))
-        return cls(analytics_providers)
+        cls.__instance = cls(analytics_providers, providers)
+        return cls.__instance
 
-    def __init__(self, providers=[]):
+    def __init__(self, providers=[], provider_strings=[]):
         self.providers = providers
+        self.provider_strings = provider_strings
```
Comment (leonardr) [inline analytics.py]: We agreed on Slack that you can don't need this and can take it out.

File: model.py [hunk 3]
```diff
@@ -3132,6 +3131,8 @@ def cover_thumbnail_url(self):
 
     @property
     def target_age_string(self):
+        if not self.target_age:
+            return None
```
Comment (leonardr) [inline model.py]: This needs to be the empty string, not None. There are a number of places where we append .target_age_string onto another string.

Comment (skomputer) [inline model.py]: @leonardr should it also return `""` instead of `None` a few lines below?

Comment (leonardr) [inline model.py]: Yes.

**Comment by leonardr** [comment]
A final :+1:

**Pull request merged**